snitun/server/run.py

```diff
@@ -291,5 +289,12 @@ def _process(self, con: socket.socket, workers_lb: Iterable[ServerWorker]) -> No
                 return
             _LOGGER.debug("No responsible worker for %s", hostname)
 
+        self._close_socket(con)
+
+    @staticmethod
+    def _close_socket(con: socket.socket, shutdown: bool = True) -> None:
```

Comment:
it's a bit weird that we have a function do 2 different things based on a boolean, shouldn't it be 2 functions instead ?

Reply:
It's the same, is only the hard clearing of resource/handler which is only needed in some cases, but normally it's shutdown + close. That make it more readable and avoid 5-6 times that code:
```
with suppress(OSError):
   shutdown
   close
```

Reply:
ah yes there is no else. This is ok.